matrix_client/api.py

```diff
@@ -836,3 +836,76 @@ def delete_devices(self, auth_body, devices):
             "devices": devices
         }
         return self._send("POST", "/delete_devices", content=content)
+
+    def upload_keys(self, device_keys={}, one_time_keys={}):
+        """Publishes end-to-end encryption keys for the device.
+
+        Said device must be the one used when logging in.
+
+        Args:
+            device_keys (dict): Optional. Identity keys for the device.
+            one_time_keys (dict): Optional. One-time public keys.
+        """
+        content = {}
+        if device_keys:
+            content["device_keys"] = device_keys
+        if one_time_keys:
+            content["one_time_keys"] = one_time_keys
+        return self._send("POST", "/keys/upload", content=content)
+
+    def query_user_keys(self, user_id):
```

Comment:
This is a higher level method that I don't think belongs in this class.

Reply:
right, same for claim_key then?

Reply:
assumed it was